components/service/glean/src/test/java/mozilla/components/service/glean/LabeledMetricTypeTest.kt

```diff
@@ -0,0 +1,388 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean
+
+import mozilla.components.service.glean.storages.BooleansStorageEngine
+import mozilla.components.service.glean.storages.CountersStorageEngine
+import mozilla.components.service.glean.storages.StringListsStorageEngine
+import mozilla.components.service.glean.storages.StringsStorageEngine
+import mozilla.components.service.glean.storages.TimespansStorageEngine
+import mozilla.components.service.glean.storages.UuidsStorageEngine
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import java.util.UUID
+
+@RunWith(RobolectricTestRunner::class)
+class LabeledMetricTypeTest {
```

Comment:
What happens when using empty strings as keys? Can we add test coverage for that? Should we also add test coverage to make sure we catch attempts to use "/" in the name?

Reply:
These should be caught at build time, so I'll probably add the tests for this in `glean_parser`.  Since the second part of `glean` is `lean`, I'd rather not add runtime checks.

Reply:
Though, I guess users could make their own metrics without going through `metrics.yaml`, but we discourage that.  (Don't know if there's any way to prevent that).  Hmmm...  Maybe we need runtime checks after all...

Reply:
Sorry -- I was thinking you were talking about the storage engine keys, not just the labels.  I've added tests for these things on labels.  I think for now, I'm comfortable with relying on the build-time checks for the other things that make up the storage engine key (name and category).

Reply:
My bad, I should have been less cryptic and much clearer :)